cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java

```diff
@@ -64,169 +65,432 @@ public List<Parameter> getParameters() {
+        // Initialize a new model for the export
+        CgmesMetadataModel modelForExport = new CgmesMetadataModel(subset, CgmesExportContext.DEFAULT_MODELING_AUTHORITY_SET_VALUE);
+        modelForExport.setProfile(context.getCim().getProfileUri(subset.getIdentifier()));
+
+        // If a model extension exists, use it as basis for the export
+        CgmesMetadataModels networkModels = network.getExtension(CgmesMetadataModels.class);
+        Optional<CgmesMetadataModel> networkSubsetModel = networkModels != null ?
+                networkModels.getModelForSubset(subset) :
+                Optional.empty();
+        networkSubsetModel.ifPresent(m -> modelForExport.setId(m.getId()));
+        networkSubsetModel.ifPresent(m -> modelForExport.setDescription(m.getDescription()));
+        networkSubsetModel.ifPresent(m -> modelForExport.setVersion(m.getVersion()));
+        networkSubsetModel.ifPresent(m -> modelForExport.addDependentOn(m.getDependentOn()));
+        networkSubsetModel.ifPresent(m -> modelForExport.addProfiles(m.getProfiles()));
+        networkSubsetModel.ifPresent(m -> modelForExport.setModelingAuthoritySet(m.getModelingAuthoritySet()));
+
+        // Use parameters if they have been defined.
+        // It doesn't apply to subnetworks, except for the version of updated SSH of a CGM export
+        if ((mainNetwork.equals(Boolean.TRUE) || modelUpdate.equals(Boolean.TRUE)) && context.getModelDescription() != null) {
+            modelForExport.setDescription(context.getModelDescription());
+        }
+        if ((mainNetwork.equals(Boolean.TRUE) || modelUpdate.equals(Boolean.TRUE)) && context.getModelVersion() != null) {
+            modelForExport.setVersion(Integer.parseInt(context.getModelVersion()));
+        }
+        if (mainNetwork.equals(Boolean.TRUE) && context.getModelingAuthoritySet() != null) {
+            modelForExport.setModelingAuthoritySet(context.getModelingAuthoritySet());
+        }
+
+        // Initialize the model id in the case of an updated SSH/SV of a CGM export, or if it isn't present
+        if (modelUpdate.equals(Boolean.TRUE) || modelForExport.getId() == null) {
+            CgmesExportUtil.initializeModelId(network, modelForExport, context);
+        }
+
+        return modelForExport;
+    }
+
+    /**
+     * Update cross dependencies between the subset models through the dependentOn relationship.
+     * The IGMs updated SSH supersede the original ones.
+     * The CGM updated SV depends on the IGMs updated SSH and on the IGMs original TP.
+     * @param igmModels For each IGM: the original SSH model, the updated SSH model and the original TP model.
+     * @param updatedCgmSvModel The SV model for the CGM.
+     */
+    private void updateDependenciesCGM(Collection<IgmModelsForCgm> igmModels, CgmesMetadataModel updatedCgmSvModel) {
+        // Each updated SSH model supersedes the original one
+        igmModels.forEach(m -> m.updatedSsh.clearDependencies());
```

Comment:
I think we must keep the dependencies of the original SSH. If we clear the dependencies, we lose the SSH dependent on the EQ that we have prepared.